### ISSUE TITLE
Implement anyOf, oneOf

### ISSUE THREAD

**Issue by nohponex** [phramework/validate#1]
``` json
{
  "oneOf": [
    {
      "type": "integer"
    },
    {
      "type": "array",
      "items": {
        "type": "integer"
      }
    }
  ]
}
```

validates both 

``` json
110
```

and 

``` json
[10, 100]
```
# in objects

But when it's used in objects

``` json
{
  "type": "object",
  "oneOf": [
    {
      "properties": {
        "code": {
          "type": "integer"
        }
      }
    },
    {
      "properties": {
        "code": {
          "type": "array",
          "items": {
            "type": "integer"
          }
        }
      }
    }
  ]
}
```

validates both

``` json
{
  "code": 100
}
```

and

``` json
{
  "code": [100, 200]
}
```

the above example can be rewritten in simpler equivalent for our purpose form:

``` json
{
  "type": "object",
  "properties": {
    "code": {
      "oneOf": [
        {
          "type": "integer"
        },
        {
          "type": "array",
          "items": {
            "type": "integer"
          }
        }
      ]
    }
  }
}
```

**Comment by nohponex** [comment]
Like this example http://stackoverflow.com/a/25033301/2255129

**Comment by nohponex** [comment]
Implemented in e78a1102f9dbf10ab5c2557c16c474381d192ce7, documentation and release pending.

**Note** that first example in "in objects" is not supported and probably will not be.
